ibis/backends/flink/compiler/core.py

```diff
@@ -46,15 +49,53 @@ class FlinkCompiler(Compiler):
 
 def translate(op: ops.TableNode) -> str:
     # TODO(chloeh13q): support translation of non-select exprs (e.g. literals)
-    ast = FlinkCompiler.to_ast(op)
-    return translate_language(ast.queries[0])
+    return translate_op(op)
 
 
 @functools.singledispatch
-def translate_language(language: DML | DDL) -> str:
-    raise com.OperationNotDefinedError(f'No translation rule for {type(language)}')
-
-
-@translate_language.register(Select)
-def _select(language: Select):
-    return language.compile()
+def translate_op(op: ops.TableNode) -> str:
+    raise com.OperationNotDefinedError(f'No translation rule for {type(op)}')
+
+
+@translate_op.register(ops.Literal)
+def _literal(op: ops.Literal) -> str:
+    value = op.value
+    dtype = op.output_dtype
+
+    if dtype.is_boolean():
+        # TODO(chloeh13q): Flink supports a third boolean called "UNKNOWN"
+        return 'TRUE' if value else 'FALSE'
+    elif dtype.is_string():
+        quoted = value.replace("'", "''").replace("\\", "\\\\")
+        return f"'{quoted}'"
+    elif dtype.is_date():
```

Comment:
Nit: Feels like it would be more logical to have this with the other datetime handling below.

Reply:
I think it's pretty normal to handle this separately, as DATE type is its own type. For example, see the implementation in the base class: https://github.com/ibis-project/ibis/blob/master/ibis/backends/base/sql/registry/literal.py#L74

Reply:
Oh, this was a total nit to just put the if block physically closer to the other datetime handling; 0 actual change. :)